packages/plugin-essentials/sources/commands/add.ts

```diff
@@ -78,9 +81,9 @@ export default class AddCommand extends BaseCommand {
       output: this.context.stdout,
     });
 
-    const target = this.peer
+    const givenTarget = this.peer
       ? suggestUtils.Target.PEER
-      : this.dev
+      : this.dev || this.preferDev
         ? suggestUtils.Target.DEVELOPMENT
         : suggestUtils.Target.REGULAR;
```

Comment:
I think it would be better to abstract the decision of which strategy to use by default to an external function, what do you think? It would give something like this:

```suggestion
        : suggestStrategy(pseudoDescriptor);
```

This would require to move this part in the `allSuggestions` block, would would allow us to avoid having to split it in two places.

Reply:
I'm not sure. `givenTarget` is one decision for all the packages, while `target` within `allSuggestions` is per package in packages list.
Also I think one long decision tree is less readable (or you have a better way to make this decision?).
I updated the code however, tell me what you think.

Reply:
I prefer to encapsulate the logic in a single function - since other people will likely want to make diffs on this part of the code, it's simpler if they can see the whole logic without looking for it too much (otherwise it'd be easy for someone new to the codebase to miss the difference between `givenTarget` and `target`, for example).

Reply:
Right. Your code is much more readable.